console/src/channel/CreateCalculated.tsx

```diff
@@ -0,0 +1,288 @@
+import { CSS } from "@/css";
+import { Layout } from "@/layout";
+import type { RendererProps } from "@/layout/slice";
+
+export interface CalculatedChannelArgs {
+  channelKey?: number;
+  channelName?: string;
+  expression?: string;
+}
+
+interface CalculatedModalRendererProps extends RendererProps {
+  args?: CalculatedChannelArgs;
+}
+
+export const CREATE_CALCULATED_LAYOUT_TYPE = "createCalculatedChannel";
+
+const SAVE_TRIGGER: Triggers.Trigger = ["Control", "Enter"];
+
+export const createCalculatedLayout: Layout.State = {
+  key: CREATE_CALCULATED_LAYOUT_TYPE,
+  type: CREATE_CALCULATED_LAYOUT_TYPE,
+  windowKey: MAIN_WINDOW,
+  name: "Channel.Create.Calculated",
+  icon: "Channel",
+  location: "mosaic",
+  tab: {
+    closable: true,
+    editable: false,
+  },
+  window: {
+    resizable: false,
+    size: { height: 600, width: 1000 },
+    navTop: true,
+    showTitle: true,
+  },
+};
+
+const schema = channel.newPayload
+  .extend({
+    name: z.string().min(1, "Name must not be empty"),
+    dataType: DataType.z.transform((v) => v.toString()),
+  })
+  .refine((v) => !v.isIndex || new DataType(v.dataType).equals(DataType.TIMESTAMP), {
+    message: "Index channel must have data type TIMESTAMP",
+    path: ["dataType"],
+  })
+  .refine((v) => v.isIndex || v.index !== 0 || v.virtual, {
+    message: "Data channel must have an index",
+    path: ["index"],
+  });
+
+export const CreateCalculatedModal = (({
+  onClose,
+  args,
+}: CalculatedModalRendererProps): ReactElement => {
+
+  const client = Synnax.use();
+
+  const [initialLoading, setInitialLoading] = useState(!!args?.channelKey);
+
+  const methods = Form.use<typeof schema>({
+    schema,
+    values: {
+      key: args?.channelKey ?? 0,
+      name: args?.channelName ?? "",
+      index: 0,
+      dataType: "float32",
+      internal: false,
+      isIndex: false,
+      leaseholder: 0,
+      rate: Rate.hz(0),
+      virtual: true,
+      expression: args?.expression ?? "result = np.array([])",
+      requires: [],
+    },
+  });
+
+  useEffect(() => {
+    const loadChannel = async () => {
+      if (!args?.channelKey || !client) return;
+      try {
+        const channel = await client.channels.retrieve(args.channelKey);
+        methods.reset({
+          ...channel,
+          dataType: channel.dataType.toString(),
+        });
+      } catch (err) {
+        console.error("Error loading channel", err);
+      } finally {
+        setInitialLoading(false);
+      }
+    };
+    void loadChannel();
+  }, [args?.channelKey, client]);
+
+  const [createMore, setCreateMore] = useState(false);
+
+  const { mutate, isPending } = useMutation({
+    mutationFn: async (createMore: boolean) => {
+      if (!methods.validate() || client == null) return;
+      const d = methods.value();
+      d.dataType = d.dataType.toString();
+
+      await client.channels.create(methods.value());
+
+      if (!createMore) onClose();
+      else
+        methods.reset({
+          key: 0,
```

Comment:
These fields are a duplication of most of the fields on line 95. Create a common object and spread it into both these locations to consolidate.